client/elements/text/sc-image.html

```diff
@@ -0,0 +1,113 @@
+<link rel="import" href="/elements/text/sc-viewer-import.html">
+
+<dom-module id="sc-image">
+  <template>
+
+    <style>
+    </style>
+    
+    <iron-meta id="meta"></iron-meta>
+
+    <iron-ajax
+        id="ajax"
+        debounce-duration="500"
+        handle-as="json"
+        loading="{{isLoading}}"
+        last-error="{{lastError}}"
+        last-response="{{imagesData}}"
+        on-response="_gotData"></iron-ajax>
+
+    <ul id="images" style="display: none;">
+        <template is="dom-repeat" items="[[imagesData]]" as="image">
+            <li><img class="image" data-url="[[_getImageUrl(image.name)]]" alt="[[image.name]]"></li>      
+        </template>
+    </ul>
+
+  </template>
+
+  <script>
+      class SCImage extends Polymer.Element {
+          static get is() {
+              return 'sc-image';
+          }
+
+          static get properties() {
+              return {
+                  division: {
+                      type: String
+                  },
+                  vol: {
+                      type: String
+                  },
+                  page: {
+                    type: Number
+                  },
+                  isLoading: {
+                      type: Boolean,
+                      value: false
+                  },
+                  lastError: {
+                      type: Object
+                  },
+                  imagesData: {
+                      type: Array
+                  },
+                  viewer: {
+                      type: Object
+                  }
+              }
+          }
+
+          show_image(detail) {
+            this.division = detail.division;
+            this.vol = detail.vol;
+            this.page = detail.page;
+            this.$.ajax.url = this._getUrl();
+            this.$.ajax.generateRequest();
+          }
+
+
+          _getUrl() {
+            return `${this.$.meta.byKey('API_ROOT')}/images/${this.division}/${this.vol}`;
+          }
+
+          _getImageUrl_d(page, modificator) {
+              if(page === 0 || page === this.imagesData.length + 1) {
+                  return;
+              }
+              const img = this.imagesData.find(v => v.page === page + modificator);
+              return `${this.$.meta.byKey('IMAGES_ROOT')}/${img.name}`;
+          }
+
+          _getImageUrl(name) {
+            return `${this.$.meta.byKey('IMAGES_ROOT')}/${name}`;
+          }
+
+
+          _gotData() {
+              setTimeout(() => {
+                  const options = {
+                    url: (e) => e.dataUrl,
+                    navbar: false,
+                    transition: false
+                  }
+                  
+                  if(this.viewer){
+                    this.viewer.update();
+                  } else {
+                    this.viewer = new Viewer(this.$.images, options);
+                  }
+                  this.viewer.index = this.page - 1;
+                  this.viewer.show();
+              }, 10)
```

Comment:
try replacing the `setTimeout(function, miliseconds)` with a `requestAnimationFrame(function)`. It turns out the latter function should be used for code that updates the UI.